### DISCUSSION TITLE
WASM Module Integration Edition

### DISCUSSION THREAD

**Discussion by mpaperno** [mpaperno/MSFSTouchPortalPlugin#31] [Announcements]
## 1.0.0.0-beta1 (July-09-2022)
* Adds support for integration with custom WASM module from the [WASimCommander project](https://github.com/mpaperno/WASimCommander) (WASimModule).
    * Get and Set "Local" variables as well as practically any other 
      [variable type](https://docs.flightsimulator.com/html/Additional_Information/Reverse_Polish_Notation.htm#Types) available in the MSFS "Gauge API".
    * Provides a listing of currently available Local variables, automatically refreshed when aircraft changes in the simulator, and on demand.
    * Option to _create_ new Local variables inside the Simulator engine.
    * Execute any Calculator Code (using RPN, see link above) from within the Simulator engine, bringing control over practically any situation.
    * Request calculated values (results from calculator code) as TP States, using the same options as current simulator variables 
      (formatting, refresh rate, categorization, etc). Results can be in numeric or string format, and data can also be formatted as strings using RPN functions.
    * All the new request types (local/other variables or calculated values) can be saved/loaded to/from configuratoin files, just like was already possible with SimVars.
* Adds support for loading input event presets from the [HubHop database](https://hubhop.mobiflight.com).
    * New Action: _Activate a Simulator Event From HubHop_ - Presents a "drill down" list of loaded events, selectable by vendor/aircraft and system. 
      **Requires WASimModule or MobiFlight WASM modules** installed (more features available with the former).
    * HubHop data can, optionally, be updated upon plugin startup and manually via a new action choice (_MSFS Plugin -> Connect & Update -> Update HubHop_).
* Added support for using numeric values in hexadecimal notation (0xNNN). Easy "BCD" values, eg. frequency 339KHz is simply `0x03390000`.
* Dynamic TP States (custom-added variables) are now sorted into their respective categories in TP selectors 
  (instead of being in a "Dynamic" cateogry or just listed after the categories). Requires TP 3.0.10 or higher.
* Added actions to _Set_ cowl flaps 1-4 to specific position.
* The list of imported sim vars updated to include new `HSI_STATION_IDENT` and fix "settable" flag on `GPS_*` variables like `GPS_APPROACH_WP_TYPE`, etc.

**NOTE** For full support of all the new features, I highly recommend **installing the accompanying WASM module** which is available below (`WASimModule`).  It it not required to use the plugin but will help provide all the new features listed above.

**To install WASM module** - Simply download the zip file and extract the contents into your MSFS _Community_ folder (so that the folder `wasimcommander-module` is in the _Community_ folder).   MSFS would need to be restarted after adding the module.

<hr /><em>This discussion was created from the release <a href='https://github.com/mpaperno/MSFSTouchPortalPlugin/releases/tag/1.0.0.0-beta1'>WASM Module Integration Edition</a>.</em>

**Comment by magicnorm** [comment]
Hi, I new to all this, looking in the "Trigger a Simulator Event from loaded HubHop data" I see that PMDG-DC-6 is supported but I don't see the PMDG 737. I know that there was/is a new SDK. Is this something that will be added or I not using it like it should?

Thanks

**Reply by mpaperno** [reply]
Thanks again for the heads-up, this is a great catch.  I started issue #32 and posted a workaround there as well. Cheers!

**Comment by mpaperno** [comment]
Hi, and hey, thanks for being the first one to finally use these discussions for what they're intended!  :-)

This looks like a bug somewhere...  it should be in the Aircraft/Device list but I don't see it either.  I'm looking into it and will get back to you!

Cheers,
-Max
